### PR TITLE
[sdk/python]: initialize optional fields to None to match deserializer

### PR DESCRIPTION
**Pull request by Jaguar0625**
problem: deserializer sets missing optional field to None but constructor sets it to default value
solution: initial optional fields with None in constructor too

### CODE REVIEW

File: sdk/javascript/generator/StructTypeFormatter.py [hunk 1]
```diff
@@ -318,7 +333,13 @@ def generate_serialize_field(self, field):
...
 			serialize_line = f'buffer.write({serialize_field});{field_comment}\n'
 
-		return indent_if_conditional(condition, serialize_line)
+		if not condition:
+			return serialize_line
+
+		serialize_field_default = field.extensions.printer.store(field.extensions.printer.get_default_value())
+		return indent_if_conditional(
+			condition,
+			f'buffer.write({check_unset(f"this.{lang_field_name(field.name)}")} ? {serialize_field_default} : {serialize_field});\n')
```
Comment (gimre-xymcity) [inline sdk/javascript/generator/StructTypeFormatter.py:342]: this part is bit weird, you have early quit above, so if we're inside `if condition` it means that most likely `condition` is already checking variable (or well condition) that's about to be serialized...
So I _think_ `check_unset` here is kinda redundant, so I _guess_ you should be able to use same `serialize_line` as above?

File: sdk/python/examples/descriptors/nem_namespace.py [hunk 2]
```diff
@@ -5,7 +5,6 @@ def descriptor_factory():
 			'type': 'namespace_registration_transaction',
 			'rental_fee_sink': 'TAMESPACEWH4MKFMBCVFERDPOOP4FK7MTDJEYP35',
 			'rental_fee': 50000 * 1000000,
-			'parent_name': '',
```
Comment (gimre-xymcity) [inline sdk/python/examples/descriptors/nem_namespace.py:8]: hmm, looking at it again, why was this change needed?

(this is nem, so serialization should result in a difference between UNSET parent_name and EMPTY parent_name)

File: sdk/javascript/generator/StructTypeFormatter.py (hunk 1 above)
Comment (Jaguar0625) [reply sdk/javascript/generator/StructTypeFormatter.py:342]: you're right, can simplify

File: sdk/python/examples/descriptors/nem_namespace.py (hunk 2 above)
Comment (Jaguar0625) [reply sdk/python/examples/descriptors/nem_namespace.py:8]: wasn't *needed* 

> this is nem, so serialization should result in a difference between UNSET parent_name and EMPTY parent_name

sdk does not support that difference, so serialization is identical for '' and undefined

File: sdk/javascript/src/nem/models.js [hunk 3]
```diff
@@ -1677,7 +1677,7 @@ class MosaicDefinition {
 		size += arrayHelpers.size(this.properties);
 		size += 4;
 		if (0 !== this.levySize)
-			size += (undefined === this.levy || null === this.levy) ? 0 : this.levy.size;
+			size += this.levy.size;
```
Comment (gimre-xymcity) [inline sdk/javascript/src/nem/models.js:1680]: this (levySize + levy) _could_ have been potentially one of the problematic places, so if it works, we're probably missing some tests, BUT

if this should be handled differently in the future it's clearly not in the scope of this change :)

**Review by gimre-xymcity — APPROVED** [review]
